internal/multierror/multierror.go

```diff
@@ -0,0 +1,44 @@
+package multierror
+
+import (
+	"fmt"
+	"strings"
+)
+
+// Join is a drop-in replacement for errors.Join with better formatting.
+func Join(errs ...error) error {
+	n := 0
+	for _, err := range errs {
+		if err != nil {
+			n++
+		}
+	}
+	if n == 0 {
+		return nil
+	}
+	e := &joinError{
+		errs: make([]error, 0, n),
+	}
+	for _, err := range errs {
+		if err != nil {
+			e.errs = append(e.errs, err)
+		}
+	}
+	return e
+}
+
+type joinError struct {
+	errs []error
+}
+
+func (e *joinError) Error() string {
+	stringErrs := make([]string, 0, len(e.errs))
+	for _, subErr := range e.errs {
+		stringErrs = append(stringErrs, strings.Replace(subErr.Error(), "\n", "\n\t", -1))
+	}
+	return fmt.Sprintf("%d errors occurred:\n\t* %s", len(stringErrs), strings.Join(stringErrs, "\n\t* "))
+}
```

Comment:
I'm wondering if we can hoist this out into a unexported function, and use it on a `JoinError` interface instead of relying on the nested `.Error()` for `JoinError`, as that way we could format nested `errors.Join` from the stdlib as long as our "special" multierror is on top.

I'll attach a patch in a minute.

Reply:
> I'm wondering if we can hoist this out into a unexported function, and use it on a JoinError interface instead of relying on the nested .Error() for JoinError, as that way we could format nested errors.Join from the stdlib as long as our "special" multierror is on top.

But then, we're going once again to debate about whether we should force the formatting of any other error implementing that interface. Plus, we don't have the case yet.

I don't want to go through this rabbit hole. For sure there's room for improvement here, but the goal is to unstuck pending PRs that rely on joined errors.